### PR TITLE
Tauri grid changes and fixes

### PR DESCRIPTION
**Pull request by Ascarbek**
### Feature Preview

In PR includes:
 - Grid Filtering
 - Grid Sorting
 - Grid column resizing
 - ui and backend fixes

fixes https://github.com/AppFlowy-IO/AppFlowy/issues/2429
fixes https://github.com/AppFlowy-IO/AppFlowy/issues/2430
fixes https://github.com/AppFlowy-IO/AppFlowy/issues/2425
fixes https://github.com/AppFlowy-IO/AppFlowy/issues/2427

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_tauri/vite.config.ts.timestamp-1683797439178-90bd485d13b3.mjs
Comment (qinluhe) [inline frontend/appflowy_tauri/vite.config.ts.timestamp-1683797439178-90bd485d13b3.mjs:1]: What is this?

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/CheckListProgress.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const CheckListProgress = ({ completed, max }: { completed: number; max:
             {completed > 0 && filledCheckListBars({ amount: completed })}
             {max - completed > 0 && emptyCheckListBars({ amount: max - completed })}
           </div>
-          <div className={'text-xs text-shade-4'}>{((100 * completed) / max).toFixed(0)}%</div>
+          <div className={'text-shade-4 text-xs'}>{((100 * completed) / max).toFixed(0)}%</div>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/CheckListProgress.tsx:10]: text-test-caption

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterItem.tsx [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+            onClick: () => onSelectLogicalOperatorClick(operatorName),
+          }))}
+          className={'fixed z-10 text-sm'}
+          style={{ top: `${logicalOperatorSelectTop}px`, left: `${logicalOperatorSelectLeft}px`, width: `${88}px` }}
+          onOutsideClick={() => setShowLogicalOperatorSelect(false)}
+        ></PopupSelect>
+      )}
+
+      {showValueOptions && currentFieldId && (
+        <PopupWindow
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterItem.tsx:387]: Maybe you can use `import Popover from @mui/material`.  I unified the style of this part, you can use it directly

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterPopup.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  // null row represents new filter
+  const rows = useMemo(() => (showBlankFilter ? filters.concat([null]) : filters), [filters, showBlankFilter]);
+
+  return (
+    <>
+      <div className={'fixed inset-0 z-10 backdrop-blur-sm'}></div>
+
+      <div className={'fixed inset-0 z-10 flex items-center justify-center overflow-y-auto'}>
+        <div className='flex flex-col rounded-lg bg-white shadow-md' ref={refContainer}>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterPopup.tsx:70]: bg-white -> bg-body

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            className={`flex w-[180px] items-center justify-between rounded-lg border px-2 py-1 ${
+              showOrderSelect ? 'border-fill-hover' : 'border-shade-4'
+            }`}
+            ref={refOrderSelect}
+            onClick={onOrderClick}
+          >
+            {currentOrder ? (
+              <SortLabel value={currentOrder}></SortLabel>
+            ) : (
+              <span className={'text-shade-4'}>Select order</span>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx:129]: text-test-caption

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+          onClick={() => onDelete?.()}
+          className={`rounded p-1 hover:bg-fill-list-hover ${data ? 'opacity-100' : 'opacity-0'}`}
+        >
+          <i className={'block h-[16px] w-[16px]'}>
+            <TrashSvg />
+          </i>
+        </button>
+      </div>
+      {showFieldSelect && (
+        <PopupSelect
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx:143]: Maybe you can use import { Select } from @mui/material

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortPopup.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  };
+
+  const rows = useMemo(() => (showBlankSort ? [...sort, null] : sort), [sort, showBlankSort]);
+
+  return (
+    <>
+      <div className={'fixed inset-0 z-10 backdrop-blur-sm'}></div>
+
+      <div className={'fixed inset-0 z-10 flex items-center justify-center overflow-y-auto'}>
+        <div className='flex flex-col rounded-lg bg-white shadow-md' ref={refContainer}>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortPopup.tsx:67]: bg-white -> bg-bg-body

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/CheckList/NewCheckListOption.tsx [hunk 7]
```diff
@@ -44,7 +44,7 @@ export const NewCheckListOption = ({
       />
       <button
         onClick={() => onSaveNewOptionClick()}
-        className={'flex items-center gap-2 rounded-lg bg-main-accent px-4 py-2 text-white hover:bg-main-hovered'}
+        className={'flex items-center gap-2 rounded-lg bg-fill-hover px-4 py-2 text-white hover:bg-main-hovered'}
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/CheckList/NewCheckListOption.tsx:47]: text-white -> text-content-onfill

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/EditCellOptionPopup.tsx [hunk 8]
```diff
@@ -110,7 +113,7 @@ export const EditCellOptionPopup = ({
           </i>
           <span>{t('grid.selectOption.deleteTag')}</span>
         </button>
-        <div className={'bg-shade-6 -mx-4 h-[1px]'}></div>
+        <div className={'bg-line-divider -mx-4 h-[1px]'}></div>
         <div className={'text-shade-3 my-2 font-medium'}>{t('grid.selectOption.colorPanelTitle')}</div>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/EditCellOptionPopup.tsx:117]: text-shade-3  -> text-text-caption

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/MultiSelectTypeOptions.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      <hr className={'-mx-2 my-2 border-line-divider'} />
+      <div className={'flex flex-col gap-1'}>
+        <div className={'flex items-center justify-between px-3 py-1.5'}>
+          <div>Options</div>
+          {!showInput && <button onClick={() => setShowInput(true)}>Add option</button>}
+        </div>
+        {showInput && (
+          <div
+            ref={inputContainerRef}
+            className={`border-shades-3 flex items-center gap-2 rounded border bg-main-selector px-2`}
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/MultiSelectTypeOptions.tsx:61]: border-shades-3 -> border-line-divider
bg-mian-selector -> bg-fill-hover/bg-fill-list-hover

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/MultiSelectTypeOptions.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                ref={inputRef}
+                style={{ width: newInputWidth }}
+                className={'py-2'}
+                value={value}
+                onChange={(e) => setValue(e.target.value)}
+                onBlur={() => setShowInput(false)}
+                onKeyDown={onKeyDown}
+              />
+            )}
+            <div className={'font-mono text-shade-3'}>{value.length}/30</div>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/Options/MultiSelectTypeOptions.tsx:74]: text-shade-3 -> text-text-caption

File: frontend/appflowy_tauri/src/appflowy_app/components/auth/SignUp/SignUp.tsx [hunk 11]
```diff
@@ -104,7 +104,7 @@ export const SignUp = () => {
             <span className='text-xs text-gray-500'>
               {t('signUp.alreadyHaveAnAccount')}
               <Link to={'/auth/login'}>
-                <span className='ml-2 text-main-accent hover:text-main-hovered'>{t('signIn.buttonText')}</span>
+                <span className='ml-2 text-fill-hover hover:text-main-hovered'>{t('signIn.buttonText')}</span>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/auth/SignUp/SignUp.tsx:107]: hover:text-content-hover

File: frontend/appflowy_tauri/src/appflowy_app/components/grid/GridCell/GridSingleSelectOptions.tsx [hunk 12]
```diff
@@ -65,6 +63,7 @@ export default function GridSingleSelectOptions({
           left={editCellOptionLeft}
           cellIdentifier={cellIdentifier}
           editingSelectOption={editingSelectOption}
+          onUpdateSelectOption={() => console.log('placeholder')}
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/grid/GridCell/GridSingleSelectOptions.tsx:66]: remove console

File: frontend/appflowy_tauri/src/appflowy_app/components/grid/GridTableHeader/GridTableHeaderItem.tsx [hunk 13]
```diff
@@ -60,66 +76,94 @@ export const GridTableHeaderItem = ({
...
-          <ChangeFieldTypePopup
-            top={changeFieldTypeTop}
-            left={changeFieldTypeRight}
-            onClick={(newType) => changeFieldType(newType)}
-            onOutsideClick={() => setShowChangeFieldTypePopup(false)}
-          ></ChangeFieldTypePopup>
-        )}
+        <div className={'flex w-full items-center justify-between py-2 pl-2'} ref={ref}>
+          <div className={'flex min-w-0 items-center gap-2'}>
+            <div className={'text-shade-3 flex h-5 w-5 flex-shrink-0 items-center justify-center'}>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/grid/GridTableHeader/GridTableHeaderItem.tsx:97]: text-text-caption

File: frontend/appflowy_tauri/src/appflowy_app/components/grid/GridTableHeader/GridTableHeaderItem.tsx [hunk 14]
```diff
@@ -60,66 +76,94 @@ export const GridTableHeaderItem = ({
...
+              <FieldTypeIcon fieldType={field.fieldType}></FieldTypeIcon>
+            </div>
+            <span className={'text-shade-3 overflow-hidden text-ellipsis whitespace-nowrap'}>{field.title}</span>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/grid/GridTableHeader/GridTableHeaderItem.tsx:100]: text-text-caption

File: frontend/appflowy_tauri/vite.config.ts.timestamp-1683797439178-90bd485d13b3.mjs
Comment (Ascarbek) [reply frontend/appflowy_tauri/vite.config.ts.timestamp-1683797439178-90bd485d13b3.mjs:1]: no idea =) deleted

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterItem.tsx (hunk 2 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseFilter/DatabaseFilterItem.tsx]: Agree, changed to Popover

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx (hunk 5 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx]: using ButtonPopoverList

File: frontend/appflowy_tauri/src/appflowy_app/components/grid/GridCell/GridSingleSelectOptions.tsx (hunk 12 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/grid/GridCell/GridSingleSelectOptions.tsx]: fixed

**Pull request merged**